### ISSUE TITLE
Error budget - Fedora Copr frontend seems 95% up only

### ISSUE THREAD

**Issue by praiskup** [fedora-copr/copr#2897]
.. per internal Prometheus reports.

**Comment by RodoMa92** [comment]
It seems to be down for me here in europe, just getting 504 Gateway Timeouts and 503 Temporarily Unavailable

**Comment by praiskup** [comment]
Thank you for the ping @RodoMa92  ... indeed, I had to restart httpd service ... weird.

**Comment by FrostyX** [comment]
I just applied this hotfix 

```diff
[root@copr-fe ~][PROD]# dnf diff copr-frontend /usr/share/copr/coprs_frontend/coprs/oidc.py
--- /usr/share/copr/coprs_frontend/coprs/oidc.py        2023-08-15 20:52:10.000000000 +0000
+++ /usr/share/copr/coprs_frontend/coprs/oidc.py        2023-09-05 14:52:16.598179636 +0000
@@ -19,6 +19,10 @@
     """
     Check whether the config is valid
     """
+
+    # Hotfixing, to avoid a lot of tracebacks in /var/log/httpd/error_log
+    return False
+
     if not is_config_valid(config):
         logger.error("OIDC_LOGIN or OIDC_PROVIDER_NAME is empty")
         return False
```

There was a huge number of tracebacks per minute. Let's see if it helps.

**Comment by praiskup** [comment]
We don't need to parallelize the queries with `max_parallel_workers` I think (these queries should be as simple as possible, so more workers are not needed).  Are the new migrations responsible for this (explain analyze)?

    daa62cd0743d_add_unique_constraint_to_fas_group.py
    08dd42f4c304_connect_copr_to_actions.py
    c6dd61c09256_buildchroot_tags.py

We also updated the system, so perhaps a new version of something?

**Comment by praiskup** [comment]
The disk load in last 3 days:
![Screenshot_20230908_071645](https://github.com/fedora-copr/copr/assets/3277839/0ee300b5-991d-4341-877d-955b74087d35)

**Comment by FrostyX** [comment]
When I EXPLAIN ANALYZE the queries now, they seem fine, e.g.

```
coprdb=# EXPLAIN ANALYZE SELECT build.id AS build_id, build.pkgs AS build_pkgs, build.built_packages AS build_built_packages, build.pkg_version AS build_pkg_version, build.canceled AS build_canceled, build.repos AS build_repos, build.submitted_on AS build_submitted_on, build.result_dir AS build_result_dir, build.memory_reqs AS build_memory_reqs, build.timeout AS build_timeout, build.enable_net AS build_enable_net, build.source_type AS build_source_type, build.source_json AS build_source_json, build.fail_type AS build_fail_type, build.is_background AS build_is_background, build.source_status AS build_source_status, build.srpm_url AS build_srpm_url, build.isolation AS build_isolation, build.bootstrap AS build_bootstrap, build.user_id AS build_user_id, build.copr_id AS build_copr_id, build.package_id AS build_package_id, build.batch_id AS build_batch_id, build.module_id AS build_module_id, build.copr_dir_id AS build_copr_dir_id, build.scm_object_id AS build_scm_object_id, build.scm_object_type AS build_scm_object_type, build.scm_object_url AS build_scm_object_url, build.update_callback AS build_update_callback, build.submitted_by AS build_submitted_by, build.resubmitted_from_id AS build_resubmitted_from_id
        FROM build
        WHERE NOT build.canceled AND (build.id IN (SELECT build_chroot.build_id
        FROM build_chroot
        WHERE build_chroot.status IN (3, 4, 6, 7, 9)) OR build.source_status IN (3, 4, 6, 7)) AND build.batch_id IS NOT NULL;
┌─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│                                                                               QUERY PLAN                                                                                │
├─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ Gather  (cost=6033.90..233856.51 rows=155955 width=654) (actual time=439.072..781.091 rows=31 loops=1)                                                                  │
│   Workers Planned: 2                                                                                                                                                    │
│   Workers Launched: 2                                                                                                                                                   │
│   ->  Parallel Seq Scan on build  (cost=5033.90..217261.01 rows=64981 width=654) (actual time=434.286..741.645 rows=10 loops=3)                                         │
│         Filter: ((NOT canceled) AND (batch_id IS NOT NULL) AND ((hashed SubPlan 1) OR (source_status = ANY ('{3,4,6,7}'::integer[]))))                                  │
│         Rows Removed by Filter: 1098652                                                                                                                                 │
│         SubPlan 1                                                                                                                                                       │
│           ->  Index Scan using build_chroot_status_started_on_idx on build_chroot  (cost=0.43..5030.22 rows=1469 width=4) (actual time=0.074..10.284 rows=3340 loops=3) │
│                 Index Cond: (status = ANY ('{3,4,6,7,9}'::integer[]))                                                                                                   │
│ Planning Time: 0.325 ms                                                                                                                                                 │
│ Execution Time: 781.147 ms                                                                                                                                              │
└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
(11 rows)
```

Is it possible that we hit some postgres deadlock making all the SELECTs just wait (see the `D` state in `htop` screenshot) which got (temporarily) resolved by me restarting the PostgreSQL service yesterday? I doubt it though.

Also, thank you for pointing out the disk load, I don't know how to explain that either.

**Comment by praiskup** [comment]
Mirek allocated more IOPS to the max value of gp3 volume type (+ ~$10/month).  Let's see if it helps.

**Comment by praiskup** [comment]
Experimenting with 3GB shared PG buffers.

**Comment by praiskup** [comment]
Fixed by 251d76f0387cc8588df621f755be89774b563e76

**Comment by praiskup** [comment]
Mirek also switched the disk to io2 in the meantime.
